pkg/tracing/options.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tracing
+
+import "github.com/spf13/cobra"
+
+// Options defines the set of options supported by Istio's component tracing package.
+type Options struct {
+	// URL of zipkin collector (example: 'http://zipkin:9411/api/v1/spans'). This enables tracing for Mixer itself.
+	ZipkinURL string
+
+	// URL of jaeger HTTP collector (example: 'http://jaeger:14268/api/traces?format=jaeger.thrift'). This enables tracing for Mixer itself.
+	JaegerURL string
+
+	// Whether or not to emit trace spans as log records.
+	LogTraceSpans bool
+}
+
+// NewOptions returns a new set of options, initialized to the defaults
+func NewOptions() *Options {
+	return &Options{}
+}
+
+// TracingEnabled returns whether the given options enable tracing to take place.
+func (o *Options) TracingEnabled() bool {
+	return o.JaegerURL != "" || o.ZipkinURL != "" || o.LogTraceSpans
+}
+
+// AttachCobraFlags attaches a set of Cobra flags to the given Cobra command.
+//
+// Cobra is the command-line processor that Istio uses. This command attaches
+// the necessary set of flags to expose a CLI to let the user control all
+// tracing options.
+func (o *Options) AttachCobraFlags(cmd *cobra.Command) {
+	cmd.PersistentFlags().StringVarP(&o.ZipkinURL, "trace_zipkin_url", "", "",
+		"URL of Zipkin collector (example: 'http://zipkin:9411/api/v1/spans').")
+
+	cmd.PersistentFlags().StringVarP(&o.JaegerURL, "trace_jaeger_url", "", "",
+		"URL of Jaeger HTTP collector (example: 'http://jaeger:14268/api/traces?format=jaeger.thrift').")
+
+	cmd.PersistentFlags().BoolVarP(&o.LogTraceSpans, "trace_log", "", false,
+		"Whether or not to log trace spans.")
+
+	// TODO: Remove all this stuff by the 0.5 release
```

Comment:
is there a TODO to also update the YAML templates ?

Reply:
I extended the TODO comment here and in the other file to remind the reader to address templates, examples, docs.